tests/test_app.py

```diff
@@ -99,6 +99,42 @@ def test_register_proposal_with_array_json(self):
         self.assertEqual(400, re.status_code)
         self.assertEqual({"error": 'payload must be a valid json'}, re.json)
 
+    def test_unregister_proposal_successful(self):
+        public_address = generate_static_public_address()
+        payload = {
+            "service_proposal": {
+                "id": 1,
+                "format": "service-proposal/v1",
+                "provider_id": public_address,
+            }
+        }
+
+        auth = generate_test_authorization(json.dumps(payload))
+        re = self._post(
+            '/v1/register_proposal',
+            payload,
+            headers=auth['headers'])
+        self.assertEqual(200, re.status_code)
+        self.assertIsNotNone(re.json)
+
+        node = Node.query.get(public_address)
+        self.assertEqual(self.REMOTE_ADDR, node.ip)
```

Comment:
`/v1/register_proposal` is already tested - instead of invoking endpoint to create service proposal create, we can create model directly.